pkg/endpoint/endpoint.go

```diff
@@ -1827,7 +1827,7 @@ func (e *Endpoint) identityLabelsChanged(ctx context.Context, myChangeRev int) (
 	// continue even if the parent has given up. Enforce a timeout of two
 	// minutes to avoid blocking forever but give plenty of time to release
 	// the identity.
-	releaseCtx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	releaseCtx, cancel := context.WithTimeout(ctx, option.Config.KVstoreConnectivityTimeout)
```

Comment:
@aanm and I had a discussion privately about reusing `ctx` here. In this particular case it's possible for the parent context to give up after the identity allocation above but before we reach one of the failure cases below. In that case, the failure handling inside `releaseNewlyAllocatedIdentity()` will log a warning, and we will rely upon the lease timeout in the kvstore to release the identity.

Looks like we need to update the code comments to make this clear but from that perspective we believe it's fine to fail out the identity release path below immediately, rather than adding an additional 2 minute delay here in violation of the parent API context cancellation request.